src/coreclr/jit/lsra.h

```diff
@@ -2493,7 +2498,7 @@ class RefPosition
     // Prior to the allocation pass, registerAssignment captures the valid registers
     // for this RefPosition.
     // After the allocation pass, this contains the actual assignment
-    regMaskTP registerAssignment;
+    SingleTypeRegSet registerAssignment;
```

Comment:
This can contain multiple different register types for `RefTypeKill`. I think there are plenty of unused fields in `RefPosition` for `RefTypeKill` to fit a union that has a full `regMaskTP`.

Is it possible to double check that the other changed register sets in this PR don't also suffer from this problem? For example, some asserts somewhere to validate that `SingleTypeRegSet` don't have both float and int registers.

Reply:
> This can contain multiple different register types for RefTypeKill

I realized that while working on this PR, but wanted to have minimal changes to keep things unblocked. I will run into this problem anyway when I add predicate registers, because now, the `registerAssignment` for `RefTypeKill` need to technically hold more than 64 registers. One of the option as you suggested is to `union` some of the unused fields for `RefTypeKill` or convert `SingleTypeRegSet` to `64RegsSet` data type (same concept but different name to identify that former just had register set of one type vs. the newer one has 64 registers in a set of mixed type), but with the later approach, we might end up creating 2 `RefTypeKill` RefPosition - first for gpr/float and second for predicate. I will see which one is better approach and go with the minimalistic change needed at this point to unblock the predicate register implementation. 

> Is it possible to double check that the other changed register sets in this PR don't also suffer from this problem

There may be and the plan is to fix them up when I add predicate register, because then we will have to differentiate between which register type we are touching.